pipelines/internal/commands/export/export.go

```diff
@@ -0,0 +1,204 @@
+// Copyright 2018 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package export provides a sub-tool for exporting pipelines to BigQuery.
+package export
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"flag"
+	"fmt"
+	"time"
+
+	"cloud.google.com/go/bigquery"
+	"cloud.google.com/go/civil"
+	genomics "google.golang.org/api/genomics/v2alpha1"
+)
+
+var (
+	flags = flag.NewFlagSet("", flag.ExitOnError)
+
+	filter      = flags.String("filter", "", "the export filter")
+	datasetName = flags.String("dataset", "", "the dataset to export to which must already exist")
+	tableName   = flags.String("table", "", "the table to export to")
+)
+
+type row struct {
+	Name  string
+	Done  bool
+	Error *status `bigquery:",nullable"`
+
+	// The raw pipeline JSON.
+	Pipeline string
+
+	Labels []label
+	Events []event
+
+	CreateTime         civil.DateTime
+	StartTime, EndTime bigquery.NullDateTime
+
+	// Additional fields pulled out of the pipeline for convenience.
+	Regions     []string
+	Zones       []string
+	MachineType string
+	Preemptible bool
+}
+
+type status struct {
+	Message string
+	Code    int64
+}
+
+type event struct {
+	Timestamp   civil.DateTime
+	Description string
+}
+
+type label struct {
+	Key, Value string
+}
+
+func Invoke(ctx context.Context, service *genomics.Service, project string, arguments []string) error {
+	flags.Parse(arguments)
+
+	if *datasetName == "" || *tableName == "" {
+		return errors.New("dataset and table are required")
+	}
+
+	path := fmt.Sprintf("projects/%s/operations", project)
+	call := service.Projects.Operations.List(path).Context(ctx)
+	call.PageSize(1024)
+
+	if *filter != "" {
+		call = call.Filter(*filter)
+	}
+
+	bq, err := bigquery.NewClient(ctx, project)
+	if err != nil {
+		return err
+	}
+
+	dataset := bq.Dataset(*datasetName)
+	if _, err := dataset.Metadata(ctx); err != nil {
+		return err
+	}
+
+	schema, err := bigquery.InferSchema(row{})
+	if err != nil {
+		return err
+	}
+
+	table := dataset.Table(*tableName)
+	if _, err := table.Metadata(ctx); err != nil {
+		if err := table.Create(ctx, &bigquery.TableMetadata{Schema: schema}); err != nil {
+			return err
+		}
+
+		// Using the table immediately after creating it sometimes causes upload
```

Comment:
Rather than sleeping for an arbitrary amount of time can we query the table status instead?

Reply:
After testing some more I can't repro the issue so I've just removed this.